### PR TITLE
Refactor to allow arbitrary optional message arguments for each data source

### PR DESCRIPTION
**Pull request by aaronsn**

### CODE REVIEW

File: airflow/dags/util.py [hunk 1]
```diff
@@ -7,42 +7,55 @@
...
+def generate_bq_payload(workflow_id: str, dataset: str, filename: str = None,
                         gcs_bucket: str = None) -> dict:
     """Creates the payload object required for the BQ ingestion operator.
 
-    filename: Name of gcs file to get the data from.
     workflow_id: ID of the datasource workflow. Should match ID defined in
                  DATA_SOURCES_DICT.
     dataset: Name of the BQ dataset to write the data to.
+    filename: Name of gcs file to get the data from.
     gcs_bucket: GCS bucket to write to. Defaults to the GCS_LANDING_BUCKET env
```
Comment (ldbnr4) [inline airflow/dags/util.py:52]: nit: "GCS bucket to read from..."

**Review by ldbnr4 — APPROVED** [review]
One small nit but other than that lgtm

File: python/datasources/data_source.py [hunk 2]
```diff
@@ -16,24 +16,33 @@ def get_table_name():
...
-            Include the file extension."""
-        url_file_to_gcs.url_file_to_gcs(url, None, gcs_bucket, filename)
+        attrs: Additional message attributes such as url and filename that are
+               needed for this data source."""
+        url_file_to_gcs.url_file_to_gcs(self.get_attr(attrs, 'url'), None,
+                                        gcs_bucket,
+                                        self.get_attr(attrs, 'filename'))
 
-    def write_to_bq(self, dataset, gcs_bucket, filename):
+    def write_to_bq(self, dataset, gcs_bucket, attrs):
```
Comment (jenniebrown) [inline python/datasources/data_source.py:36]: Should this be **attrs?

File: run_gcs_to_bq/main.py [hunk 3]
```diff
@@ -67,7 +61,7 @@ def do_ingestion(event):
         raise RuntimeError("ID: {}, is not a valid id".format(workflow_id))
 
     data_source = DATA_SOURCES_DICT[workflow_id]
-    data_source.write_to_bq(dataset, gcs_bucket, filename)
+    data_source.write_to_bq(dataset, gcs_bucket, **attributes)
```
Comment (jenniebrown) [inline run_gcs_to_bq/main.py:64]: nit: **attrs for consistency?

**Review by jenniebrown — CHANGES_REQUESTED** [review]

File: python/datasources/data_source.py (hunk 2 above)
Comment (aaronsn) [reply python/datasources/data_source.py:36]: Good catch

**Review by jenniebrown — APPROVED** [review]

**Pull request merged**